packages/react/src/hooks/__tests__/useTheme.test.ts

```diff
@@ -0,0 +1,46 @@
+import { createTheme, defaultTheme } from '@aws-amplify/ui';
+import * as React from 'react';
+import { useTheme } from '../useTheme';
+
+jest.mock('react');
+
+describe('useTheme', () => {
+  afterEach(() => jest.clearAllMocks());
+
+  it('should return a theme object', () => {
+    const customTheme = createTheme({
+      name: 'my-theme',
+      tokens: {
+        colors: {
+          font: {
+            primary: { value: 'red' },
+          },
+        },
+      },
+    });
+
+    (React.useContext as jest.Mock).mockReturnValue({ theme: customTheme });
+
+    const theme = useTheme();
```

Comment:
It would be better to use the `renderHook` helper from `@testing-library/react-hooks` to test custom hooks. We are already doing that in other hooks test cases.

Reply:
good idea! will refactor to use `renderHook`